app/components/views/TicketsPage/MyTicketsTab/MyTicketsTab.jsx

```diff
@@ -1,120 +1,140 @@
-      {
-        key: "immature",
-        value: { status: "immature" },
-        label: labels.immature
-      },
-      { key: "live", value: { status: "live" }, label: labels.live },
-      { key: "voted", value: { status: "voted" }, label: labels.voted },
-      { key: "missed", value: { status: "missed" }, label: labels.missed },
-      { key: "expired", value: { status: "expired" }, label: labels.expired },
-      { key: "revoked", value: { status: "revoked" }, label: labels.revoked }
-    ];
-  }
+const selectTicketTypeFromFilter = (filter) => {
+  const ticketType = ticketTypes.find(
+    (ticket) => filter.status === ticket.value.status
+  );
+  return ticketType && ticketType.key;
+};
 
-  selectedTicketTypeFromFilter(filter) {
-    const types = this.getTicketTypes();
-    let key;
-    types.forEach((type) => {
-      if (filter.status === type.value.status) {
-        key = type.key;
-        return;
-      }
-    });
-    return key;
-  }
+const MyTickets = (props) => {
+  const {
+    tickets,
+    tsDate,
+    noMoreTickets,
+    ticketsFilter,
+    window,
+    goBackHistory,
+    getTickets,
+    changeTicketsFilter
+  } = useTicketsList();
 
-  onChangeFilter(value) {
-    const newFilter = {
-      ...this.props.ticketsFilter,
-      ...value
-    };
-    this.props.changeTicketsFilter(newFilter);
-  }
+  const [selectedTicketTypeKey, setTicketTypeKey] = useState(
+    selectTicketTypeFromFilter(ticketsFilter)
+  );
+  const [selectedSortOrderKey, setSortOrderKey] = useState(
+    ticketsFilter.listDirection
+  );
 
-  onChangeSelectedType(type) {
-    this.onChangeFilter(type.value);
-    this.setState({ selectedTicketTypeKey: type.key });
-  }
+  const onChangeFilter = (filter) => {
+    const newFilter = { ...ticketsFilter, ...filter };
+    changeTicketsFilter(newFilter);
+  };
 
-  onChangeSortType(type) {
-    this.onChangeFilter({ listDirection: type.value });
-    this.setState({ selectedSortOrderKey: type.value });
-  }
+  const onChangeSelectedType = (type) => {
+    onChangeFilter(type.value);
+    setTicketTypeKey(type.key);
+  };
 
-  render() {
-    const loadMoreThreshold =
-      90 + Math.max(0, this.props.window.innerHeight - 765);
-    return (
-      <TicketListPage
-        {...{
-          ...this.props,
-          ...this.state,
-          loadMoreThreshold,
-          ticketTypes: this.getTicketTypes(),
-          sortTypes: this.getSortTypes(),
-          tickets: this.props.tickets,
-          ...substruct(
-            {
-              onChangeSelectedType: null,
-              onChangeSortType: null
-            },
-            this
-          )
-        }}
-      />
-    );
-  }
-}
+  const onChangeSortType = (type) => {
+    onChangeFilter({ listDirection: type.value });
+    setSortOrderKey(type.value);
+  };
+
+  const loadMoreThreshold = 90 + Math.max(0, window.innerHeight - 765);
+
+  return (
+    <TicketListPage
+      {...{
+        ...props,
```

Comment:
this isn't needed anymore, the props in this case was only the connector files, you can ditch it